src/net/sourceforge/kolmafia/session/BanishManager.java

```diff
@@ -162,312 +228,233 @@ public static final void loadBanishedMonsters() {
+        "banishedMonsters",
+        banishedMonsters.stream()
+            .flatMap(
+                m -> Stream.of(m.getMonsterName(), m.getBanisher().getName(), m.getTurnBanished()))
+            .map(Object::toString)
+            .collect(Collectors.joining(":")));
+  }
 
+  /**
+   * Iterate through all banished monsters removing entries if the supplied predicate matches.
+   *
+   * @param predicate Predicate dictating removal
+   */
+  private static final void resetIf(Predicate<BanishedMonster> predicate) {
+    BanishManager.banishedMonsters.removeIf(predicate);
     BanishManager.saveBanishedMonsters();
   }
 
-  public static final void resetAvatar() {
+  /**
+   * Iterate through all banished monsters removing entries if the supplied predicate matches their
+   * Reset type.
+   *
+   * @param predicate Predicate dictating removal
+   */
+  private static final void resetIfType(Predicate<Reset> predicate) {
+    resetIf(m -> predicate.test(m.getBanisher().getResetType()));
+  }
 
-    BanishManager.banishedMonsters.removeIf(
-        current ->
-            BanishManager.findBanisher(current.getBanishName()).getResetType()
-                == Reset.AVATAR_RESET);
+  public static final void resetRollover() {
+    resetIfType(Reset::isRolloverReset);
+  }
 
-    BanishManager.saveBanishedMonsters();
+  public static final void resetAvatar() {
+    resetIfType(r -> r == Reset.AVATAR_RESET);
   }
 
   public static final void resetAscension() {
-
-    BanishManager.banishedMonsters.removeIf(
-        current ->
-            BanishManager.findBanisher(current.getBanishName()).getResetType()
-                != Reset.NEVER_RESET);
-
-    BanishManager.saveBanishedMonsters();
+    resetIfType(r -> r != Reset.NEVER_RESET);
   }
 
-  public static final void update() {
-    BanishManager.recalculate();
-    BanishManager.saveBanishedMonsters();
+  public static final void resetCosmicBowlingBall() {
+    resetIfType(r -> r == Reset.COSMIC_BOWLING_BALL_RESET);
   }
 
-  private static void recalculate() {
-    Iterator<BanishedMonster> it = BanishManager.banishedMonsters.iterator();
-
-    while (it.hasNext()) {
-      BanishedMonster current = it.next();
-      int banisherDuration = BanishManager.findBanisher(current.getBanishName()).getDuration();
-      Reset resetType = BanishManager.findBanisher(current.getBanishName()).getResetType();
-      if ((resetType == Reset.TURN_RESET || resetType == Reset.TURN_ROLLOVER_RESET)
-          && (current.getTurnBanished() + banisherDuration <= KoLCharacter.getCurrentRun())) {
-        it.remove();
-      }
-    }
+  public static void recalculate() {
+    resetIf(Predicate.not(BanishedMonster::isValid));
   }
 
-  private static final Banisher findBanisher(final String banisher) {
-    for (Banisher ban : BANISHER) {
-      if (ban.getName().equals(banisher)) {
-        return ban;
-      }
-    }
-    return null;
+  public static final void banishCurrentMonster(final String banisherName) {
```

Comment:
if this isn't used anywhere, you could delete it to keep the code using the type-safe version